src/main/java/io/nitric/api/queue/Queue.java

```diff
@@ -53,50 +52,59 @@
  *  var payload = Map.of("orderId", orderId, "serialNumber", serialNumber);
  *  var task = Task.build(payload);
  *
- *  // Send a task to the 'shipping' queue client
- *  var client = QueueClient.build("shipping");
- *  client.send(task);
+ *  // Send a task to the 'shipping' queue
+ *  var queue = Queues.queue("shipping");
+ *  queue.send(task);
  *
  *  // Receive a list of tasks from the 'shipping' queue
- *  List&lt;Task&gt; tasks = client.receive(100);
+ *  List&lt;Task&gt; tasks = queue.receive(100);
  * </code></pre>
  *
- * @see Task
- * @see FailedTask
- *
- * @since 1.0
+ * @see Queues
  */
-public class QueueClient {
+public class Queue {
+
+    final String name;
 
-    final String queue;
-    final QueueGrpc.QueueBlockingStub serviceStub;
+    // Constructor ------------------------------------------------------------
 
     /*
-     * Enforce builder pattern.
+     * Enforce package builder patterns.
      */
-    QueueClient(Builder builder) {
-        this.queue = builder.queue;
-        this.serviceStub = builder.serviceStub;
+    Queue(String name) {
+        Contracts.requireNonBlank(name, "name");
+        this.name = name;
     }
 
     // Public Methods ---------------------------------------------------------
 
+    /**
+     * @return the queue name
+     */
+    public String getName() {
+        return name;
+    }
+
     /**
      * Send the given task to the client queue.
      *
      * @param task the task to send to the queue (required)
      */
     public void send(Task task) {
```

Comment:
Might be good to consider rolling send and sendBatch together, then we can look at deprecating and removing the singular method on the membrane side. We can still present the singular method SDK side, but have it utilise the batch implementation.

Reply:
Yeah, that makes sense. 

The ```sendBatch``` method is also good in terms of signaling failed send, we should probably add this to the signature of the ```sendTask``` method.

```
public FailedTask sendTask(Task)
```